File: Java/Java EE/Servlets/FirstServlet/src/Model/entity/abstracts/Toy.java
package Model.entity.abstracts;

public class Toy {
    private double price;
    private double weight;
    private double size;

    public Toy() {
    }

    public Toy(double price, double weight, double size) {
        this.price = price;
        this.weight = weight;
        this.size = size;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public double getSize() {
        return size;
    }

    public void setSize(double size) {
        this.size = size;
    }

    @Override
    public String toString() {
        return new StringBuilder().append("Params:\n\tPrice: ").append(price).append("\n\tSize: ").append(size).append("\n\tWeight: ").append(weight).toString();//price,size,weight
    }
}

File: Csharp/Lab8/Lab8/Lab8/cng/Model/Entity/Abstracts/Toy.cs
using System;
using System.Text;

namespace Lab8.cng.Model.Entity.Abstracts
{
    public class Toy : IComparable<Toy>
    {

        private double _price;
        private double _weight;
        private double _size;

        public Toy() {
        }

        public Toy(double price, double weight, double size) {
            this._price = price;
            this._weight = weight;
            this._size = size;
        }

        public double GetPrice() {
            return _price;
        }

        public void SetPrice(double price) {
            this._price = price;
        }

        public double GetWeight() {
            return _weight;
        }

        public void SetWeight(double weight) {
            this._weight = weight;
        }

        public double GetSize() {
            return _size;
        }

        public void SetSize(double size) {
            this._size = size;
        }


        public override int GetHashCode()
        {
            const int hashconst = 31;
            int result = 1;
            result = hashconst * result + (int)_price;
            result = hashconst * result + (int)_weight;
            result = hashconst * result + (int)_size;
            return result;
        }


        public override bool Equals(object obj)
        {
            if (obj != null && obj.GetType() == this.GetType() && obj.GetHashCode() == this.GetHashCode()){
                return true;
            }else{
                return false;
            }
        }

        
        public override String ToString() {
            return new StringBuilder().Append("Params:\n\tPrice: ").Append(_price).Append("\n\tSize: ").Append(_size).Append("\n\tWeight: ").Append(_weight).ToString();//price,size,weight
        }

        public int CompareTo(Toy obj)
        {
            return (int) (this.GetPrice() - obj.GetPrice());
        }
    }
}